httpx/_exceptions.py

```diff
@@ -59,22 +61,28 @@ class ProxyError(HTTPError):
 # HTTP exceptions...
 
 
-class ProtocolError(HTTPError):
+class ProtocolError(RequestError):
     """
     Malformed HTTP.
     """
 
 
-class DecodingError(HTTPError):
+class DecodingError(RequestError):
     """
     Decoding of the response failed.
     """
 
 
+class HTTPStatusError(RequestError):
+    """
+    HTTP Status Error.
```

Comment:
Probably worth being a bit more precise about *when* this might be raised?

```suggestion
    Called `.raise_for_status()` on an HTTP response with an error status code.
```